Fix save logic by moving setter into loop grabbing the value

mikejolley opened the pull request:
Broken by https://github.com/woocommerce/woocommerce/pull/22650 phpcs changes here https://github.com/woocommerce/woocommerce/commit/5b3b285a9d6c2e409bc16116d51c1df369aa1747

Looks like the intention of that change was to avoid setting the POST global by storing local variables. But due to this being done over 2 loops, only the last `$value` was saved.

This PR fixes that by moving the setter into the validation/formatting loop then saving once validation is done. 

Note I removed the global customer session set which syncs data because of https://github.com/woocommerce/woocommerce/pull/22392 which means this is no longer needed.

mikejolley committed:
Fix save logic by moving setter into loop grabbing the value

Broken by https://github.com/woocommerce/woocommerce/pull/22650 phpcs changes

peterfabian reviewed (commented):
Saving address works fine after the change. 

One thing I noticed, while maybe not directly changed right now: setting an invalid email address ends up with a fatal error, which it shouldn't I think. Should I create a separate issue or would you look into it here?

claudiosanches committed:
Catch WC_Data_Exception to avoid black screen

Also incldued a check to avoid duplicated messages about invalid billing
address emails.

claudiosanches reviewed (approved):
@mikejolley @peterfabian 

Peter is correct. It's we need to catch `WC_Data_Exception`' exceptions:

```
PHP Fatal error:  Uncaught WC_Data_Exception: Invalid billing email address in /var/www/woo/wp-content/plugins/woocommerce/includes/abstracts/abstract-wc-data.php:794
```

I included a fix in ff614f4f99c92cf8a24928bcc5a9383131620502

Now should all works fine.
